src/rpc/blockchain.cpp

```diff
@@ -90,7 +91,7 @@ UniValue blockheaderToJSON(const CBlockIndex* blockindex)
 UniValue blockToJSON(const CBlock& block, const CBlockIndex* blockindex, bool txDetails = false)
 {
     UniValue result(UniValue::VOBJ);
-    result.push_back(Pair("hash", block.GetHash().GetHex()));
+    result.push_back(Pair("hash", blockindex->GetBlockHash().GetHex()));
```

Comment:
GetHash() calculates the hash, GetBlockHash returns a previously-calculated hash.  So it's an efficiency improvement.